src/routes/formsg/formsgSiteLaunch.ts

```diff
@@ -350,6 +356,35 @@ export class FormsgSiteLaunchRouter {
               `Unable to get dig response for domain: ${launchResult.value.primaryDomainSource}. Skipping check for AAAA records`
             )
           }
+
+          if (!caaDigResponse) {
+            logger.info(
+              `Unable to get dig response for domain: ${launchResult.value.primaryDomainSource}. Skipping check for CAA records`
+            )
+          } else if (caaDigResponse.answer) {
+            const caaRecords = caaDigResponse.answer
+
+            /**
+             * NOTE: If there exists more than one CAA Record, we need to
+             * 1. check if they have whitelisted Amazon CAA
+             * 2. if not, send email to inform them to whitelist Amazon CAA
+             */
+            const hasAmazonCAAWhitelisted = caaRecords.some((record) => {
+              const isAmazonCAA = TRUSTED_AMPLIFY_CAA_RECORDS.some(
+                (trustedCAA) => trustedCAA === record.value
+              )
+              return isAmazonCAA
+            })
+            if (caaRecords.length > 1 && !hasAmazonCAAWhitelisted) {
+              successResult.addCAARecord = true
+            } else {
+              successResult.addCAARecord = false
```

Comment:
don't we need the `else if asAmazonCAAWhitelisted` check here?

Reply:
oh my conidtion is wrong i think that caused the confusion ps

Reply:
changed, hopefully this clearer